src/v/s3/client.cc

```diff
@@ -443,6 +513,46 @@ ss::future<http::client::response_stream_ref> client::get_object(
       });
 }
 
+ss::future<client::head_object_result> client::head_object(
+  bucket_name const& name,
+  object_key const& key,
+  const ss::lowres_clock::duration& timeout) {
```

Comment:
nit: taking this duration type by reference probably doesn't provide any benefit. the underlying numeric value is probably just as big as the pointer itself?